### ISSUE TITLE
How to get version number without releasing? 

### ISSUE THREAD

**Issue by mariusGundersen** [mgoltzsche/conventional-release#6]
I would love to get the version number without releasing, so that I can embed the version number in the build, then when the build has succeeded I want to create a release with the build output.

**Comment by mgoltzsche** [comment]
You can get the version using the environment variable `RELEASE_VERSION` within job steps that follow the conventional-release Action invocation. This env var is always populated. In case of a non-release build the version contains a `-dev-<SHA>` suffix.
Do you only want to disable the GitHub release creation or do you also want to disable the git tag creation?
To disable the GitHub release creation, you can set the [`github-release-enabled` option](https://github.com/mgoltzsche/conventional-release/blob/v0.7.5/action.yml#L27) to `false` - a git tag will still be published this way, though.

**Comment by mgoltzsche** [comment]
Actually, this GitHub Action is already designed to generate the `RELEASE_VERSION` and a temporary git tag without publishing it yet before running your other steps within the job afterwards. Only after all other steps you defined in your workflow job that called the conventional-release Action ran successfully, the Action runs a post-entrypoint that finally publishes the git tag and creates a GitHub release for it, see [here](https://github.com/mgoltzsche/conventional-release?tab=readme-ov-file#example-workflow), allowing you to build your app within the same job using the version number.
In other words: the Action invokation does not publish a release yet but a release may be published at the end of a successful job that invoked the Action initially.
Thus, unless you really want to manually test your app before releasing it, there is no need to script the release yourself - the Action already takes care of that and still allows you to run your build and tests within the job the usual GitHub Actions way prior to publishing the git tag (and GitHub release).

**Comment by mariusGundersen** [comment]
So if the steps are

1. Create release
2. Build project

And the build fails then the release won't be published?

**Comment by mgoltzsche** [comment]
yes, exactly! In that case the conventional-release Action will neither push the git tag nor create the GitHub release.

**Comment by mgoltzsche** [comment]
It works like that because the action implements both a regular and a post-entrypoint (that runs at the end of the job): The regular entrypoint generates the version and creates the corresponding git tag locally while the post-entrypoint finally pushes the tag and creates the GitHub release only when the [job was successful](https://github.com/mgoltzsche/conventional-release/blob/v0.7.5/action.yml#L43).

**Comment by mgoltzsche** [comment]
I am closing the issue, assuming your problem is solved.
Please feel free to reopen if that's not the case.

**Comment by mariusGundersen** [comment]
Yes, thank you, it works exactly as you described. Maybe it should be better documented in the Readme?
